scipy/spatial/setlist.pxd

```diff
@@ -47,6 +47,14 @@ cdef inline int init(setlist_t *setlist, size_t n, size_t size_guess) except -1:
         setlist.sizes[j] = 0
         setlist.alloc_sizes[j] = size_guess
         setlist.sets[j] = <int*>libc.stdlib.malloc(sizeof(int) * size_guess)
+        if setlist.alloc_sizes == NULL:
+            # free all sets[i]
+            for i in range(j):
+                libc.stdlib.free(<void*>setlist.sets[i])
```

Comment:
This is almost a direct copy of `free()` directly after this.
it feels weird to pass `<void*>` but it's what `free()` does so I copied assuming I don't understand something.

Reply:
No problem.  It probably isn't necessary, but maybe some compilers complain if it isn't used?